File: service/src/main/java/hu/bsstudio/robonaut/race/skill/SkillRaceService.java
package hu.bsstudio.robonaut.race.skill;

import hu.bsstudio.robonaut.race.skill.model.GateInformation;
import hu.bsstudio.robonaut.race.skill.model.SkillRaceResult;
import hu.bsstudio.robonaut.team.model.DetailedTeam;
import reactor.core.publisher.Mono;

public interface SkillRaceService {
    Mono<DetailedTeam> updateSkillRaceResultOnGate(final GateInformation gateInformation);

    Mono<DetailedTeam> updateSkillRaceResult(final SkillRaceResult skillRaceResult);
}

File: server/service/src/main/kotlin/hu/bsstudio/robonaut/race/skill/SkillRaceService.kt
package hu.bsstudio.robonaut.race.skill

import hu.bsstudio.robonaut.race.skill.model.GateInformation
import hu.bsstudio.robonaut.race.skill.model.SkillRaceResult
import hu.bsstudio.robonaut.team.model.DetailedTeam
import reactor.core.publisher.Mono

interface SkillRaceService {
  fun updateSkillRaceResultOnGate(gateInformation: GateInformation): Mono<DetailedTeam>

  fun updateSkillRaceResult(skillRaceResult: SkillRaceResult): Mono<DetailedTeam>
}
